src/realm/views.cpp

```diff
@@ -34,14 +34,15 @@ SortDescriptor::SortDescriptor(Table const& table, std::vector<std::vector<size_
                                std::vector<bool> ascending)
     : m_ascending(std::move(ascending))
 {
-    REALM_ASSERT(!column_indices.empty());
     REALM_ASSERT_EX(m_ascending.empty() || m_ascending.size() == column_indices.size(), m_ascending.size(),
                     column_indices.size());
     if (m_ascending.empty())
         m_ascending.resize(column_indices.size(), true);
-    if (table.is_degenerate())
-        return;
-
+    if (table.is_degenerate()) {
+        // We need access to the column acessors and that's not available in a
+        // degenerate table, it would crash if we continued to try and access it
+        throw LogicError(LogicError::descriptor_on_degenerate_table);
```

Comment:
Is this a degenerate *subtable"? Sorting a degenerate subtable must not throw, it must simply be a no-op. A degenerate subtable must behave like an ordinary fully backed 0-rows subtable, even though its representation is a nullpointer. It's not an error to sort it